allensdk/core/typing.py

```diff
@@ -0,0 +1,9 @@
+from typing import _Protocol
```

Comment:
I think you can do `from typing_extensions import Protocol` instead.

Didn't know about this before, really cool feature!
See: https://mypy.readthedocs.io/en/latest/protocols.html

Reply:
since typing is a standard library I prefer to get it from there rather than another dependency. Starting python 3.8 it's included in `typing`